scripts/tests/run_tv_casting_test.py

```diff
@@ -63,6 +64,21 @@ def __exit__(self, exception_type, exception_value, traceback):
         self.process.wait()
 
 
+def remove_cached_files(cached_file_pattern: str) -> bool:
+    """Remove any cached files that match the provided pattern."""
+
+    cached_files = glob.glob(cached_file_pattern)  # Returns a list of paths that match the pattern.
+
+    for cached_file in cached_files:
+        try:
+            os.remove(cached_file)
+        except OSError as e:
+            logging.error(f'Failed to remove cached file `{cached_file}` with error: `{e.strerror}`.')
+            return False
```

Comment:
the flow seems to be: call this bool function, which does a printout of the exception and the caller exits(1).

Why not directly throw the exception and not make this return a bool? If you want to handle the exception you can catch it, however turning an exception into a bool and then using it to sys exit with error feels like just extra complexity.

If you want to add file name context, you can re-throw the exception with more context, however it should still be an exception.

Reply:
Thank you for the feedback, I have addressed this in the latest commit.